vscode/src/editor/utils/folding-ranges.ts

```diff
@@ -0,0 +1,228 @@
+import * as vscode from 'vscode'
+
+import { getFoldingRanges, getSymbols } from '.'
+
+/**
+ * Gets the folding range containing the target position.
+ *
+ * NOTE: Use getSmartSelection from utils/index.ts instead
+ *
+ * @param uri - The URI of the document to get folding ranges for
+ * @param target - The target position number
+ * @returns The folding range containing the target position, or undefined if none found
+ */
+export async function getTargetFoldingRange(uri: vscode.Uri, target: number): Promise<vscode.Selection | undefined> {
+    // Check if symbols is available in the doc by its file uri and look for language id
+    const doc = await vscode.workspace.openTextDocument(uri)
+    // Remove text-based languages that don't support symbols or folding ranges
+    const textBaseLangIds = ['markdown', 'plaintext', 'json', 'sql']
+    const isSupported = !textBaseLangIds.includes(doc.languageId)
+
+    // Get the ranges of all classes and folding ranges in parallel
+    const [ranges, classes] = await Promise.all([
+        getFoldingRanges(uri).then(r => r?.filter(r => !r.kind)),
+        isSupported
+            ? getSymbols(uri)
+                  .then(r => r?.filter(s => s.kind === vscode.SymbolKind.Class))
+                  .then(s => s?.map(symbol => symbol.location.range))
+            : [],
+    ])
+
+    // If lanaguage id not available, find class object ranges heuristically
+    const classRanges = isSupported ? classes : await getOuterClassFoldingRanges(ranges, uri)
+
+    const targetRange = getNonClassOutermostFoldingRanges(classRanges, ranges, target, isSupported)
```

Comment:
```suggestion
    const isPlainText = !isSupported
    const targetRange = getNonClassOutermostFoldingRanges(classRanges, ranges, target, isPlainText)
```

Fundamentally the idea is that `isSupported` is the inverse of `isPlainText`. You wanted an isPlainText check to combine neighbour ranges for plaintext documents as shown [here](https://github.com/sourcegraph/cody/pull/1163/files#diff-7a994af9d659344fecd4e2771e864229c0cf1158c4ce5acbe08957b906cf765aR172) 

But the inverse was missed in the document flow so now the `combineNeiborFoldingRanges` actually works for NON text based files(I.E. Normal files like typescript etc)  which I don't think was your intention.

Reply:
OMG yes this was where i messed up! Thanks for finding this i was wondering why it wasn't working anymore 😆 

#### Before your change

Doc added on top of the wrong folding range:

<img width="776" alt="image" src="https://github.com/sourcegraph/cody/assets/68532117/9589528b-89d8-495d-99c2-9b9620151683">


#### After your change

Doc is added at the correct folding range

<img width="942" alt="image" src="https://github.com/sourcegraph/cody/assets/68532117/663733b7-7900-4590-a713-9982d7fdd4f4">

THANK YOU!